homeassistant/components/discovery/__init__.py

```diff
@@ -59,7 +59,7 @@
     SERVICE_ENIGMA2: ("media_player", "enigma2"),
     SERVICE_WINK: ("wink", None),
     SERVICE_SABNZBD: ("sabnzbd", None),
-    SERVICE_SAMSUNG_PRINTER: ("sensor", "syncthru"),
+    SERVICE_SAMSUNG_PRINTER: ("sensor", None),
```

Comment:
You can remove this whole line. There is no need to load the `sensor` integration when printer is detected.
```suggestion
```

Reply:
But if I do _that_, then I get a `Unknown service discovered: samsung_printer {...}` on every startup from discovery, even when the device has already been configured. I think this would be confusing to users and should be avoided.

What I had previously (entries in SERVICE_HANDLERS and MIGRATED_SERVICE_HANDLERS) seemed to do exactly the exact desired thing: no confusing messages, and no unwanted loading of the platform. I think I should revert back to that.

Reply:
That's what `MIGRATED_SERVICE_HANDLERS` is for :)

Reply:
Also lowered log level to `debug` in #37617

Reply:
I'm somewhat confused now -- earlier you noted about the entry I added in MIGRATED_SERVICE_HANDLERS:

| You can remove it from here. The migrated ones are if they have step_discovery, but we won't add those anymore in favor of the manifest based discovery.

...so should I add it back now?